### PR TITLE
IGNITE-16857 Java thin: Add AtomicLong

### PR DESCRIPTION
**Pull request by ptupitsyn**
* Add `IgniteClient.atomicLong()` and `ClientAtomicLong` APIs.
* Provides the same functionality as "thick" API, with the only exception of `AtomicConfiguration.affinityFunction`.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/client/ClientAtomicLong.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ * </ul>
+ * <p>
+ * <h1 class="header">Creating Distributed Atomic Long</h1>
+ * Instance of distributed atomic long can be created by calling the following method:
+ * <ul>
+ *     <li>{@link IgniteClient#atomicLong(String, long, boolean)}</li>
+ * </ul>
+ * @see IgniteClient#atomicLong(String, long, boolean)
+ */
+
```
Comment (isapego) [inline modules/core/src/main/java/org/apache/ignite/client/ClientAtomicLong.java:62]: Is this empyline here intended?

File: modules/core/src/main/java/org/apache/ignite/client/ClientAtomicLong.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    /**
+     * Gets status of atomic.
+     *
+     * @return {@code true} if atomic was removed from cache, {@code false} in other case.
+     */
+    public boolean removed();
+
+    /**
+     * Removes this atomic long.
+     *
```
Comment (isapego) [inline modules/core/src/main/java/org/apache/ignite/client/ClientAtomicLong.java:149]: Non-needed empty line

**Review by isapego — CHANGES_REQUESTED** [review]
Overall looks good. See my minor comments inline

File: modules/core/src/main/java/org/apache/ignite/client/ClientAtomicConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+ */
+public class ClientAtomicConfiguration {
+    /** */
+    public static final int DFLT_BACKUPS = 1;
+
+    /** */
+    public static final CacheMode DFLT_CACHE_MODE = PARTITIONED;
+
+    /** Default atomic sequence reservation size. */
+    public static final int DFLT_ATOMIC_SEQUENCE_RESERVE_SIZE = 1000;
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/client/ClientAtomicConfiguration.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/client/ClientAtomicConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+    /**
+     * @return Number of backup nodes.
+     */
+    public int getBackups() {
+        return backups;
+    }
+
+    /**
+     * Constructor.
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/client/ClientAtomicConfiguration.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/client/ClientAtomicConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    /** */
+    public static final int DFLT_BACKUPS = 1;
+
+    /** */
+    public static final CacheMode DFLT_CACHE_MODE = PARTITIONED;
+
+    /** Default atomic sequence reservation size. */
+    public static final int DFLT_ATOMIC_SEQUENCE_RESERVE_SIZE = 1000;
+
+    /** Default batch size for all cache's sequences. */
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/client/ClientAtomicConfiguration.java:38]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/client/ClientAtomicLong.java (hunk 1 above)
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/client/ClientAtomicLong.java:62]: Fixed.

**Pull request merged**